### PR TITLE
WebCryptoAPI: Use .any.js for generateKey tests and split up slow tests

### LINKED ISSUE #10794: /WebCryptoAPI/generateKey/* are disabled/flaky in Mozilla Chromium WebKit
http://bocoup.github.io/wpt-disabled-tests-report/

Investigate what's up with this test:

Path | Products | Results | Bugs
-- | -- | -- | --
/WebCryptoAPI/generateKey/test_successes_RSA-OAEP.https.html | mozilla chromium webkit | disabled [ Pass Timeout ] [ Pass Failure ] | https://bugzilla.mozilla.org/show_bug.cgi?id=1279799 https://crbug.com/787971 None

Edit: also:

* /WebCryptoAPI/generateKey/test_successes_RSA-PSS.https.html
* /WebCryptoAPI/generateKey/test_successes_RSASSA-PKCS1-v1_5.https.html
* /WebCryptoAPI/generateKey/successes_RSA-OAEP.https.worker.html
* /WebCryptoAPI/generateKey/successes_RSA-PSS.https.worker.html
* /WebCryptoAPI/generateKey/successes_RSASSA-PKCS1-v1_5.https.worker.html

### PR DESCRIPTION
**Pull request by zcorpan**
Slow tests test 10 subtests at a time, using the `// META: variant=...`
feature.

Fixes #10794.

### CODE REVIEW

**Comment by Hexcles** [comment]
I like the use of `.any.js`.

The splitting is done with variants. The indexes are hard coded, which might be a bit brittle. I'm not sure how to improve though. One idea I have is to not hard code the last index, but use something like `150-last` instead to run all the remaining subtests, which also has the downside of potentially making the last split too large if people don't pay attention when adding more subtests later.

**Comment by zcorpan** [comment]
Thanks, good idea about `last`, now implemented.

**Comment by jimsch** [comment]
@zcorpan Do we have any idea if there are more or fewer failures than before.  I am not seeing anything in the changes that triggers to me that we will have more failures, but trying to look through the travis logs it seems that there are a great number of fails.

**Comment by zcorpan** [comment]
Comparing https://w3c-test.org/WebCryptoAPI/generateKey/ and https://w3c-test.org/submissions/10984/WebCryptoAPI/generateKey/

Test | Chrome, master | Chrome, this PR | Firefox, master | Firefox, this PR
--|--|--|--|--
failures_AES-CBC | 736 Pass 2 Fail | 736 Pass 2 Fail | 738 Fail | 738 Fail
successes_RSA-OAEP | 156 Pass | 156 Pass | 156 Pass | 156 Pass
successes_RSA-PSS | 36 Pass | 36 Pass | 36 Pass | 36 Pass
successes_RSASSA-PKCS1-v1_5 | 36 Pass | 36 Pass | 36 Pass | 36 Pass

(This is unfortunately manual collection of results, see https://github.com/w3c/web-platform-tests/issues/7475)

In Firefox, the all-fail in `failures_AES-CBC` is because of `assert_equals: Bad algorithm not supported expected "NotSupportedError" but got "SyntaxError"` i.e. wrong exception being thrown.

File: WebCryptoAPI/generateKey/successes.js [hunk 1]
```diff
@@ -70,6 +70,18 @@ function run_test(algorithmNames, slowTest) {
         }, testTag + ": generateKey" + parameterString(algorithm, extractable, usages));
     }
 
+    // Only test a subset of tests with, e.g., ?1-10 in the URL
+    var subTestStart = 0;
+    var subTestEnd = Infinity;
+    var match;
+    if (location.search) {
+        match = /^\?(\d+)-(\d+|last)$/.exec(location.search);
+        subTestStart = match[1];
```
Comment (jimsch) [inline WebCryptoAPI/generateKey/successes.js]: What happens if the match operation fails here?

Comment (zcorpan) [reply WebCryptoAPI/generateKey/successes.js]: I made it run all tests.

**Review by jimsch — APPROVED** [review]
Things look good.  Thanks for the change

**Comment by zcorpan** [comment]
Thanks for the reviews!

**Pull request merged**